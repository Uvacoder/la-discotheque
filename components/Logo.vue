<template>
  <div class="Logo">
    <nuxt-link
      to="/"
      @click.native="onClick">
      <h1>La Discothèque</h1>
      <img
        src="/img/svg/logo-notfinal.svg"
        alt="Logo - La Discothèque">
    </nuxt-link>
  </div>
</template>

<script>
export default {
  computed: {
    isMobile() {
      return this.$store.state.isMobile
    }
  },
  methods: {
    onClick(e) {
      if (this.isMobile) {
        this.$store.commit('SET_SIDEBAR_VISIBLE', false)
      }

      return true;
    }
  }
}
</script>

<style>
.Logo {
  width: 102px;
  height: 102px;
}

.Logo a {
  color:inherit;
}

.Logo h1 {
  visibility: hidden;
  height: 0px;
}
</style>
