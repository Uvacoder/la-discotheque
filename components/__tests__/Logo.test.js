import { shallowMount } from '@vue/test-utils'

import NuxtLink from 'nuxt/lib/app/components/nuxt-link'
import Logo from '~/components/Logo.vue'

describe('Logo.vue', () => {
  beforeEach(() => {
    Logo.components = Object.assign(Logo.components || {}, {
      NuxtLink: NuxtLink
    })
  })

  it('renders nuxt-link correctly', () => {
    const wrapper = shallowMount(Logo)
    expect(wrapper.find('a'))
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(Logo)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
