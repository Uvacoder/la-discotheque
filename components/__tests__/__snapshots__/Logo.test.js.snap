// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo.vue matches snapshot 1`] = `
<div role="banner" class="Logo">
  <nuxtlink-stub to="/" arialabel="Homepage" aria-label="Homepage">
    <h1>La Discothèque</h1> <img src="/img/svg/logo-notfinal.svg" alt="Logo - La Discothèque">
  </nuxtlink-stub>
</div>
`;
